nomad/search_endpoint_test.go

```diff
@@ -1932,6 +1979,52 @@ func TestSearch_FuzzySearch_CSIVolume(t *testing.T) {
 	require.False(t, resp.Truncations[structs.Volumes])
 }
 
+func TestSearch_FuzzySearch_HostVolume(t *testing.T) {
+	ci.Parallel(t)
+
+	srv, cleanup := TestServer(t, func(c *Config) {
+		c.NumSchedulers = 0
+	})
+	defer cleanup()
+	codec := rpcClient(t, srv)
+	testutil.WaitForLeader(t, srv.RPC)
+
+	store := srv.fsm.State()
+	index, _ := store.LatestIndex()
+
+	node := mock.Node()
+	index++
+	must.NoError(t, store.UpsertNode(structs.MsgTypeTestSetup, index, node))
+
+	id := uuid.Generate()
+	index++
+	err := store.UpsertHostVolumes(index, []*structs.HostVolume{{
+		ID:        id,
+		Name:      "example",
+		Namespace: structs.DefaultNamespace,
+		PluginID:  "glade",
+		NodeID:    node.ID,
+		NodePool:  node.NodePool,
+	}})
+	must.NoError(t, err)
+
+	req := &structs.FuzzySearchRequest{
+		Text:    id[0:3], // volumes are prefix searched
```

Comment:
this doesn't quite line up with my conception of "fuzzy", but looks like others are like this too.  I suppose the "fuzziness" is that it could match different types of things, not just a prefix search on volumes specifically.

Reply:
Right for a lot of other objects it really is fuzzy and will match on fields other than ID. But for CSI volumes we did it only by ID so I'm reproducing that behavior here. It might be worth revisiting to see whether we can expand the prefix search for volumes and other objects more widely in general.